### DISCUSSION TITLE
ISO Airgap Install on existing Oracle Linux 9 instance issue

### DISCUSSION THREAD

**Discussion by craigsmooth** [Security-Onion-Solutions/securityonion#11601] [2.4] [answered]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

airgap

### Hardware Specs

Meets minimum requirements

### CPU

4

### RAM

32

### Storage for /

3TB

### Storage for /nsm

1TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I am somewhat confused about how to go about doing an airgap install on my own build of Oracle Linux 9.2. If I mount the iso, and then run "so-setup iso" and select "airgap", install starts and I get errors almost as if it is expecting salt to already be installed. I get errors such as:

/home/xxxxx/SecurityOnion/salt "no such file or directory"
salt-call "command not found"

I don't see where salt is a prereq in the install documentation. It only mentions installing git as a prereq.
https://docs.securityonion.net/en/2.4/installation.html#manual-installation-via-other-iso-image

Am I going about this incorrectly?
Is it possible to do an ISO based Airgap install on my own installed Oracle Linux 9.2 distribution?

Thank you for any help.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by TotieBash** [comment, marked as answer]
Use the SO iso image..
https://github.com/Security-Onion-Solutions/securityonion/blob/2.4/main/DOWNLOAD_AND_VERIFY_ISO.md

The link you sent is for network install.  In other words, if you are not using the SO iso image, you can use your RHEL clone of choice and do "so-setup-network"

"Is it possible to do an ISO based Airgap install on my own installed Oracle Linux 9.2 distribution?" I believe it is possible after all its all Linux. Granted that you do all the heavy work in reverse engineer the prereq by comparing the SO image but I would imagine it would not be supported by the SO team.

**Reply by craigsmooth** [reply]
Thank you @TotieBash . With the network install, is it possible to do an airgap installation. In other words, can I do the network install which will pull down all the containers and then use a local repo. I see documentation that says and ISO install with Airgap is a thing but I don't see any mention of a network install that can be used airgapped.

**Reply by TOoSmOotH** [reply]
You can do a network install and then change it to airgap mode afterwards.

**Reply by craigsmooth** [reply]
Thank you @TOoSmOotH . Are there steps documented somewhere on how to do that?

**Reply by dougburks** [reply]
You may be able to make this work but please keep in mind that the only officially supported option is to use our Security Onion ISO image:
https://docs.securityonion.net/en/2.4/os.html
https://docs.securityonion.net/en/2.4/airgap.html

**Comment by TotieBash** [comment]
> You can do a network install and then change it to airgap mode afterwards.

hmmm I learned something new....  It might be useful later... 

@TOoSmOotH does it mean for example I can use RHEL9 or Rocky9 network install and then I can change it to airgap mode later?

**Reply by dougburks** [reply]
You may be able to make this work but please keep in mind that the only officially supported option is to use our Security Onion ISO image:
https://docs.securityonion.net/en/2.4/os.html
https://docs.securityonion.net/en/2.4/airgap.html
